projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+import { useSelect, useDispatch } from '@wordpress/data';
+import restApi from '@automattic/jetpack-api';
+import { getProductCheckoutUrl } from '@automattic/jetpack-components';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_ID } from '../../state/store.jsx';
+
+const { registrationNonce, apiRoot, apiNonce } = window?.JP_CONNECTION_INITIAL_STATE
+	? window.JP_CONNECTION_INITIAL_STATE
+	: {};
+
+/**
+ * Custom hook that performs the needed steps
+ * to concrete the checkout workflow.
+ *
+ * @param {object} props             - The props passed to the hook.
+ * @param {string} props.productSlug - The WordPress product slug.
+ * @param {string} props.siteSuffix  - The site suffix.
+ * @param {string} props.redirectUrl - The URI to redirect to after checkout.
+ * @returns {Function}				 - The useEffect hook.
+ */
+export default function useProductCheckoutWorkflow( {
+	productSlug,
+	siteSuffix,
```

Comment:
Since the js-connection package already has a initial state of its own, we could add the siteSuffix to the initial state so we don't need to pass it here

Reply:
the problem is we are not exposing values to the client in the jetpack-connection pkg. its 100% JS if I'm not wrong.

Reply:
just add it here: https://github.com/Automattic/jetpack/blob/master/projects/packages/connection/src/class-initial-state.php

but it's ok to do this in a follow up PR